File: backend/shared/src/main/java/com/devfactory/processautomation/qa/rwa/service/RegionService.java
package com.devfactory.processautomation.qa.rwa.service;

import com.devfactory.processautomation.qa.rwa.service.dtos.ItemListResponse;
import com.devfactory.processautomation.qa.rwa.service.dtos.RegionDto;

public interface RegionService {
    ItemListResponse<RegionDto> getAllRegions();
}

File: frontend/src/app/services/region.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { HttpWrapperService } from './httpWrapper.service';
import { ItemListResponse } from '../model/item-list-response';
import { Region } from '../model/region';

@Injectable({
    providedIn: 'root'
})
export class RegionService {
    constructor(private httpWrapperService: HttpWrapperService) { }

    public getAllRegions(): Observable<ItemListResponse<Region>> {
        return this.httpWrapperService.get(`regions`);
    }
}
